public/app/features/dashboard-scene/scene/ReloadOnFiltersChangeBehavior.ts

```diff
@@ -0,0 +1,38 @@
+import { SceneObjectBase, SceneObjectState, VariableDependencyConfig } from '@grafana/scenes';
+import { appEvents } from 'app/core/core';
+import { ReloadDashboardEvent } from 'app/types/events';
+
+export interface ReloadOnFiltersChangeBehaviorState extends SceneObjectState {
+  reloadOnFiltersChange?: string[] | boolean;
+  uid?: string;
+}
+
+export class ReloadOnFiltersChangeBehavior extends SceneObjectBase<ReloadOnFiltersChangeBehaviorState> {
+  public constructor(state: ReloadOnFiltersChangeBehaviorState) {
+    super(state);
+
+    if (state.reloadOnFiltersChange && state.uid) {
+      this._variableDependency = new VariableDependencyConfig(
+        this,
+        state.reloadOnFiltersChange === true
+          ? {
+              onAnyVariableChanged: this.reloadDashboard.bind(this),
```

Comment:
Assuming there's a chained variables set in the dashboard, this will trigger the reload multiple times, right @torkelo ?

Reply:
yes, and if you have the refresh on time range enabled + refresh on variable it will trigger 1 reload for time range + 1 for each variable